### PR TITLE
Enable use of self-signed certs with OCI DNS

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Separate cert-management from DNS configuration
- Self-signed and ACME issuers are available in all DNS configurations (although it may not be useful in some)
- Simplifies platform oprator 
- Enable use in OCI DNS tests by default, with options to run with ACME staging or production 

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by wmhopkins — DISMISSED** [review]
lgtm.

**Review by gilbode — APPROVED** [review]
this looks ok, but can you still supply your own cert?  or i guess could you ever supply your own & if so can you still?

File: ci/oke-ocidns/Jenkinsfile [hunk 1]
```diff
@@ -46,6 +47,10 @@ pipeline {
         choice (description: 'Kubernetes Version for OKE Cluster', name: 'OKE_CLUSTER_VERSION',
                 // 1st choice is the default value
                 choices: [ "v1.18.10", "v1.17.13", "v1.16.15" ])
+        choice (description: 'Certificate Issuer', name: 'CERT_MANAGER',
+                choices: certIssuers)
```
Comment (pfmackin) [inline ci/oke-ocidns/Jenkinsfile:51]: shouldn't self-signed be a choice?

Comment (pfmackin) [inline ci/oke-ocidns/Jenkinsfile:51]: Does it make more sense to call this CERT_ISSUER?

File: platform-operator/controllers/verrazzano/installjob/install_config.go [hunk 2]
```diff
@@ -206,105 +193,64 @@ func newOCIDNSInstallConfig(vz *installv1alpha1.Verrazzano) (*InstallConfigurati
...
+		return Certificate{
+			IssuerType: CertIssuerTypeAcme,
+			ACME: &CertificateACME{
+				Provider:     string(vz.Spec.Components.CertManager.Certificate.Acme.Provider),
+				EmailAddress: vz.Spec.Components.CertManager.Certificate.Acme.EmailAddress,
+				Environment:  vz.Spec.Components.CertManager.Certificate.Acme.Environment,
 			},
+		}
+	}
+	return Certificate{
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/installjob/install_config.go:247]: If certManager is defined in the CR with no ACME, is this the default Certificate?

File: tests/e2e/config/scripts/process_oci_dns_install_yaml.sh [hunk 3]
```diff
@@ -4,15 +4,20 @@
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 
 INSTALL_CONFIG_TO_EDIT=$1
-ACME_ENV=${2:-"staging"}
+CERT_MGR=${2:-"acme"}
```
Comment (pfmackin) [inline tests/e2e/config/scripts/process_oci_dns_install_yaml.sh:7]: see my previous comment, maybe this should be CERT_ISSUER

**Review by pfmackin — DISMISSED** [review]
Few comments, no blockers

File: platform-operator/controllers/verrazzano/installjob/install_config.go (hunk 2 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/installjob/install_config.go:247]: Yes, if the ACME struct is empty, we just default to the self-signed issuer.  Those are pretty much our only choices right?

File: tests/e2e/config/scripts/process_oci_dns_install_yaml.sh (hunk 3 above)
Comment (michael-cico) [reply tests/e2e/config/scripts/process_oci_dns_install_yaml.sh:7]: OK, I'll change this on the next pass.

File: ci/oke-ocidns/Jenkinsfile (hunk 1 above)
Comment (michael-cico) [reply ci/oke-ocidns/Jenkinsfile:51]: Fixed and pushed

**Review by pfmackin — APPROVED** [review]

**Pull request merged**